spec/datetimeformat.html

```diff
@@ -144,29 +136,26 @@
       </p>
 
       <emu-alg aoid="ToDateTimeOptions">
-        1. If _options_ is *undefined*, then let _options_ be *null*, else let _options_ be ToObject(_options_).
-        1. ReturnIfAbrupt(_options_).
+        1. If _options_ is *undefined*, let _options_ be *null*; otherwise let _options_ be ? ToObject(_options_).
         1. Let _options_ be ObjectCreate(_options_).
         1. Let _needDefaults_ be *true*.
         1. If _required_ is *"date"* or *"any"*,
           1. For each of the property names *"weekday"*, *"year"*, *"month"*, *"day"*:
             1. Let _prop_ be the property name.
-            1. Let _value_ be Get(_options_, _prop_).
-            1. ReturnIfAbrupt(_value_).
-            1. If _value_ is not *undefined*, then let _needDefaults_ be *false*.
+            1. Let _value_ be ? Get(_options_, _prop_).
+            1. If _value_ is not *undefined*, let _needDefaults_ be *false*.
         1. If _required_ is *"time"* or *"any"*,
           1. For each of the property names *"hour"*, *"minute"*, *"second"*:
             1. Let _prop_ be the property name.
-            1. Let _value_ be Get(_options_, _prop_).
-            1. ReturnIfAbrupt(_value_).
-            1. If _value_ is not *undefined*, then let _needDefaults_ be *false*.
+            1. Let _value_ be ? Get(_options_, _prop_).
+            1. If _value_ is not *undefined*, let _needDefaults_ be *false*.
         1. If _needDefaults_ is *true* and _defaults_ is either *"date"* or *"all"*, then
           1. For each of the property names *"year"*, *"month"*, *"day"*:
-            1. Let _status_ be CreateDatePropertyOrThrow(_options_, _prop_, *"numeric"*).
+            1. Let _status_ be the result of calling CreateDatePropertyOrThrow(_options_, _prop_, *"numeric"*).
```

Comment:
```
1. Let _f_ be Something().
```

it is a call to an abstract operation with an implicit step that sets `f` to the Completion Record's `[[Value]]`, while:

```
1. Let _f_ be the result of calling | evaluating Something().
```

is a call to an abstract operation that return Completion Records explicitly.

I see a lot of spec text that are using the `Let _status_ ... ` vs `Let _f_ ... ` indistinctly, because it was very hard to deal with the value vs status of the operation to abrupt of use the value, but now that we have ?, it is really simply. I think we can now enforce that when someone is using `ReturnIfAbrupt()` explicitly, it will have to be on the result of calling/evaluating step.

That's my interpretation of this topic:
http://tc39.github.io/ecma262/#sec-implicit-completion-values

/cc @bterlson can you chime in here?